core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -240,10 +279,37 @@ func (l *logPollerWrapper) LatestEvents() ([]evmRelayTypes.OracleRequest, []evmR
 		}
 	}
 
-	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "nextBlock", nextBlock, "latest", latest)
+	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "startBlock", startBlock, "endBlock", latest)
 	return resultsReq, resultsResp, nil
 }
 
+func (l *logPollerWrapper) FilterPreviouslyDetectedEvents(logs []logpoller.Log, detectedEvents *detectedEvents, filterType string) []logpoller.Log {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	filteredLogs := []logpoller.Log{}
+	for _, log := range logs {
```

Comment:
I'm a tiny bit worried about potential perf problem if logs if very long. Realistically, it's only 50 or so blocks so the total number won't be ever bigger than 1000 or so... But maybe it makes sense to add a sanity check (another const limit of 2000?) and truncate oldest ones + log errors just in case. For example if an RPC client goes crazy and dumps multiple copies of the same log?

Reply:
Do you know if the logPoller enforces ordering of events? Should I trucate from the front or the back of the returned array? I can investigate further if you are unsure.

Reply:
Pretty sure it does but please confirm. We should truncate oldest ones.

Reply:
I confirmed it does enforce ordering from oldest to newest.